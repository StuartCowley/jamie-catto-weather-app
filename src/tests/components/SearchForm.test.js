import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe('SearchForm', () => {
    const emptyFunc = () => {};

    it('renders correctly', () => {
        const { asFragment } = render(
            <SearchForm searchText=""
            setSearchText={emptyFunc}
            onSubmit={emptyFunc}/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the correct button text', () => {
        const { getByTestId } = render(
            <SearchForm searchText=""
            setSearchText={emptyFunc}
            onSubmit={emptyFunc}/>
        );
        expect(getByTestId('search__btn')).toHaveTextContent('Search');
    });
});