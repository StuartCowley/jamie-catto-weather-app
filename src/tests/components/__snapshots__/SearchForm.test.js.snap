// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="Search..."
      type="text"
      value=""
    />
    <button
      data-testid="search__btn"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
